File: 2581-DividePlayersIntoTeamsOfEqualSkill/2581-DividePlayersIntoTeamsOfEqualSkill.cpp
// Last updated: 6/11/2026, 11:21:55 AM
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        map<long long, int> mp;

        int n = skill.size();

        long long ts = (accumulate(skill.begin(), skill.end(), 0))/(n/2);

        int p = 0;

        long long ans=0;

        for(int i=0; i<n; i++){
            long long el = skill[i];
            if(mp[ts-el]>0){
                mp[ts-el] -= 1;
                if(mp[ts-el]<0){
                    mp[ts-el] =0;
                }
                ans += (ts-el)*(el);
                p +=1;
            } else {
                mp[el] += 1; 
            }
        }

        if(p == (n/2)){
            return ans;
        } else{
            return -1;
        }
    }
};

File: 2581-DividePlayersIntoTeamsOfEqualSkill/2581-DividePlayersIntoTeamsOfEqualSkill.py
# Last updated: 6/11/2026, 11:21:50 AM
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        totsum = sum(skill)
        n = len(skill)

        ps = totsum/(n/2)

        map = {}
        p = 0
        ans = 0
        for i in skill:
            if (ps-i) in map:
                map[ps-i] -= 1
                p += 1
                ans += (i*(ps-i))
                if(map[ps-i] == 0):
                    del map[ps-i]
            else:
                if i in map:
                    map[i] += 1
                else:
                    map[i] = 1
        
        if(p == n/2):
            return int(ans)
        else:
            return -1